Mathlib/Topology/VectorBundle/Riemannian.lean

```diff
@@ -0,0 +1,241 @@
+their scalar product is continuous. Version with `ContinuousAt`. -/
+lemma ContinuousAt.inner_bundle
+    (hv : ContinuousAt (fun m ↦ (v m : TotalSpace F E)) x)
+    (hw : ContinuousAt (fun m ↦ (w m : TotalSpace F E)) x) :
+    ContinuousAt (fun b ↦ ⟪v b, w b⟫) x := by
+  simp only [← continuousWithinAt_univ] at hv hw ⊢
+  exact ContinuousWithinAt.inner_bundle hv hw
+
+/-- Given two continuous maps into the same fibers of a continuous Riemannian bundle,
+their scalar product is continuous. Version with `ContinuousOn`. -/
+lemma ContinuousOn.inner_bundle
+    (hv : ContinuousOn (fun m ↦ (v m : TotalSpace F E)) s)
+    (hw : ContinuousOn (fun m ↦ (w m : TotalSpace F E)) s) :
+    ContinuousOn (fun b ↦ ⟪v b, w b⟫) s :=
+  fun x hx ↦ (hv x hx).inner_bundle (hw x hx)
+
+/-- Given two continuous maps into the same fibers of a continuous Riemannian bundle,
+their scalar product is continuous. -/
+lemma Continuous.inner_bundle
+    (hv : Continuous (fun m ↦ (v m : TotalSpace F E)))
+    (hw : Continuous (fun m ↦ (w m : TotalSpace F E))) :
+    Continuous (fun b ↦ ⟪v b, w b⟫) := by
+  simp only [continuous_iff_continuousAt] at hv hw ⊢
+  exact fun x ↦ (hv x).inner_bundle (hw x)
+
+end Continuous
+
+namespace Bundle
+
+section Construction
+
+variable
+  {B : Type*} [TopologicalSpace B]
+  {F : Type*} [NormedAddCommGroup F] [NormedSpace ℝ F]
+  {E : B → Type*} [TopologicalSpace (TotalSpace F E)]
+  [∀ b, TopologicalSpace (E b)] [∀ b, AddCommGroup (E b)] [∀ b, Module ℝ (E b)]
+  [∀ b, IsTopologicalAddGroup (E b)] [∀ b, ContinuousConstSMul ℝ (E b)]
+  [FiberBundle F E] [VectorBundle ℝ F E]
+
+open Bornology
+
+variable (E) in
+/-- A family of inner product space structures on the fibers of a fiber bundle, defining the same
+topology as the already existing one. One should more often use `ContinuousRiemannianMetric`
+or `ContMDiffRiemannianMetric` to guarantee continuity or smoothness of the scalar product as a
+function of the base point.
+
+This structure is used through `RiemannianBundle` for typeclass inference, to register the inner
+product space structure on the fibers without creating diamonds. -/
+structure RiemannianMetric where
+  /-- The scalar product along the fibers of the bundle. -/
+  inner (b : B) : E b →L[ℝ] E b →L[ℝ] ℝ
+  symm (b : B) (v w : E b) : inner b v w = inner b w v
+  pos (b : B) (v : E b) (hv : v ≠ 0) : 0 < inner b v v
+  /-- The continuity at `0` is automatic when `E b` is isomorphic to a normed space, but since
+  we are not making this assumption here we have to include it. -/
+  continuousAt (b : B) : ContinuousAt (fun (v : E b) ↦ inner b v v) 0
+  isVonNBounded (b : B) : IsVonNBounded ℝ {v : E b | inner b v v < 1}
+
+/-- `Core structure associated to a family of inner products on the fibers of a fiber bundle. This
+is an auxiliary construction to endow the fibers with an inner product space structure without
+creating diamonds. -/
+@[reducible] noncomputable def RiemannianMetric.toCore (g : RiemannianMetric E) (b : B) :
```

Comment:
InnerProductSpace.Core has a warning
"Warning: Do not use this `Core` structure if the space you are interested in already has a norm
instance defined on it, otherwise this will create a second non-defeq norm instance!"

Do you think having an analogous one here would be useful?